src/deviceDetailsChannelTableModel.cpp

```diff
@@ -0,0 +1,717 @@
+}
+
+/**
+* Adds a node to the table model. Doesn't check for duplicates or correct order.
+* @param audioClusterNode The node to add to this model.
+*/
+void DeviceDetailsChannelTableModelPrivate::addNode(la::avdecc::controller::model::AudioClusterNode const& audioClusterNode, uint16_t audioClusterNodeChannel, avdecc::ConnectionInformation const& connectionInformation)
+{
+	Q_Q(DeviceDetailsChannelTableModel);
+
+	q->beginInsertRows(QModelIndex(), _nodes.size(), _nodes.size());
+	_nodes.push_back(TableRowEntry(audioClusterNode, audioClusterNodeChannel, connectionInformation));
+	q->endInsertRows();
+}
+
+/**
+* Removes all nodes from the table model.
+*/
+void DeviceDetailsChannelTableModelPrivate::removeAllNodes()
+{
+	Q_Q(DeviceDetailsChannelTableModel);
+	q->beginResetModel();
+	_nodes.clear();
+	q->endResetModel();
+}
+
+/**
+* Gets the model data at a specific row index.
+*/
+TableRowEntry const& DeviceDetailsChannelTableModelPrivate::tableDataAtRow(int row) const
+{
+	return _nodes.at(row);
+}
+
+/**
+* Gets all user edits. 
+* @return The edits in a QMap. The key indicates the node the edit has been made on, the key of the interal QMap is the type of data changed by the user.
+*/
+QMap<la::avdecc::entity::model::DescriptorIndex, QMap<DeviceDetailsChannelTableModelColumn, QVariant>*> DeviceDetailsChannelTableModelPrivate::getChanges() const
+{
+	return _hasChangesMap;
+}
+
+/**
+* Resets all changes. That have been made by the user.
+*/
+void DeviceDetailsChannelTableModelPrivate::resetChangedData()
+{
+	Q_Q(DeviceDetailsChannelTableModel);
+	q->beginResetModel();
+	qDeleteAll(_hasChangesMap);
+	_hasChangesMap.clear();
+	q->endResetModel();
+}
+
+/**
+* Gets the row count of the table.
+*/
+int DeviceDetailsChannelTableModelPrivate::rowCount() const
+{
+	return static_cast<int>(_nodes.size());
+}
+
+/**
+* Gets the column count of the table.
+*/
+int DeviceDetailsChannelTableModelPrivate::columnCount() const
+{
+	return 3;
+}
+
+/**
+* Gets the data of a table cell.
+*/
+QVariant DeviceDetailsChannelTableModelPrivate::data(QModelIndex const& index, int role) const
+{
+	auto const column = static_cast<DeviceDetailsChannelTableModelColumn>(index.column());
+	if (role == Qt::TextAlignmentRole)
+	{
+		return Qt::AlignTop;
+	}
+	switch (column)
+	{
+		/*case DeviceDetailsChannelTableModelColumn::ChannelNumber:
+		if (role == Qt::FontRole) {
+			QFont font;
+			return font;
+		}
+		else if (role == Qt::DisplayRole) {
+			return _nodes.at(index.row()).first.descriptorIndex;
+		}*/
+		case DeviceDetailsChannelTableModelColumn::ChannelName:
+		{
+			bool hasChanges = false;
+			if (_hasChangesMap.contains(_nodes.at(index.row()).audioClusterNode.descriptorIndex) && _hasChangesMap.value(_nodes.at(index.row()).audioClusterNode.descriptorIndex)->contains(DeviceDetailsChannelTableModelColumn::ChannelName))
+			{
+				hasChanges = true;
+			}
+			if (role == Qt::FontRole && hasChanges)
+			{
+				QFont font;
+				font.setItalic(true);
```

Comment:
Italic text for pending changes I s not part of Hive at this time, we must decide if this is something we would like to generalize at some point.

Reply:
We have removed usage of italic fonts as change indication in the latest commits and use enabling/disabling the apply/revert buttons to indicate presence of local changes instead.